test/unit/Cardano/Wallet/Api/JsonSpec.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.Api.JsonSpec (spec) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Arbitraries
+    ()
+import Cardano.Wallet.Api.Types.Amount
+import Cardano.Wallet.Api.Types.CurrencyUnit
+import Cardano.Wallet.Api.Types.Percentage
+import Cardano.Wallet.Api.Types.Wallet
+import Cardano.Wallet.Api.Types.WalletAddressPoolGap
+import Cardano.Wallet.Api.Types.WalletBalance
+import Cardano.Wallet.Api.Types.WalletDelegation
+import Cardano.Wallet.Api.Types.WalletDelegationStatus
+import Cardano.Wallet.Api.Types.WalletId
+import Cardano.Wallet.Api.Types.WalletName
+import Cardano.Wallet.Api.Types.WalletPassphraseInfo
+import Cardano.Wallet.Api.Types.WalletState
+import Cardano.Wallet.Api.Types.WalletStateStatus
+
+import Data.Aeson
+    ( FromJSON, ToJSON, decode, eitherDecode, encode )
+import Data.ByteString.Lazy
+    ( ByteString )
+import Data.Either
+    ( isRight )
+import Test.Hspec
+    ( Expectation, Spec, describe, it, shouldBe, shouldSatisfy )
+import Test.QuickCheck
+    ( property )
+import Text.RawString.QQ
+    ( r )
+
+spec :: Spec
+spec = do
+    describe "can perform basic JSON deserialization" $
+        it "Wallet" $
+            (eitherDecode exampleWallet :: Either String Wallet)
```

Comment:
I don't think this matches how the API will be used. The test should be the other way around.
```
exampleWalletBytes <- B8.readFile "example-wallet.json"
encode exampleWallet `shouldBe` exampleWalletBytes
```

The wallet objects that users put and post have less fields, and will want to have decoding tests.

Reply:
For testing, we'll need:

- Golden tests for every types representation in JSON (as @rvl just described, this will ensure that we are at least aware of breaking changes). Ideally, we could _generate_ golden tests from arbitrary instances to have several ones for each Api types.
- JSON roundtrips 
- HttpApiData roundtrips (when it make sense)
- Swagger schema comparison with the specification

(not necessarily all in this PR though!)

Reply:
It's correct that this wouldn't normally be used by a user. However, having this test helps us to write Haskell clients against this API (which may occur as part of a future test suite).

Reply:
> The wallet objects that users put and post have less fields, and will want to have decoding tests.

Agreed! We can include these in the next PR.